src/controllers/csi/provisioner/controller.go

```diff
@@ -156,28 +157,38 @@ func (provisioner *OneAgentProvisioner) updateAgentInstallation(ctx context.Cont
 	}
 	latestProcessModuleConfig = latestProcessModuleConfig.AddHostGroup(dk.HostGroup())
 
+	var agentUpdater *agentUpdater
 	if dk.CodeModulesImage() != "" {
-		connectionInfo, err := dtc.GetConnectionInfo()
+		latestProcessModuleConfig = latestProcessModuleConfig.AddConnectionInfo(dk.ConnectionInfo())
+		agentUpdater, err = newAgentImageUpdater(ctx, provisioner.fs, provisioner.apiReader, provisioner.path, provisioner.recorder, dk, dynakubeMetadata.ImageDigest)
 		if err != nil {
-			log.Error(err, "error when getting OneAgent connectionInfo")
+			log.Error(err, "error when setting up the agent image updater")
+			return nil, false, err
+		}
+	} else {
+		agentUpdater, err = newAgentUrlUpdater(ctx, provisioner.fs, dtc, provisioner.path, provisioner.recorder, dk)
+		if err != nil {
+			log.Info("error when setting up the agent url updater", "error", err.Error())
 			return nil, false, err
 		}
-		latestProcessModuleConfig = latestProcessModuleConfig.AddConnectionInfo(connectionInfo)
 	}
 
 	latestProcessModuleConfigCache = newProcessModuleConfigCache(latestProcessModuleConfig)
 
-	agentUpdater, err := newAgentUpdater(ctx, provisioner.fs, provisioner.apiReader, dtc, provisioner.path, provisioner.recorder, dk)
 	if err != nil {
 		log.Info("error when setting up the agent updater", "error", err.Error())
 		return nil, false, err
 	}
-	if updatedVersion, err := agentUpdater.updateAgent(dynakubeMetadata.LatestVersion, dynakubeMetadata.TenantUUID, latestProcessModuleConfigCache); err != nil {
+	if updatedVersion, err := agentUpdater.updateAgent(dynakubeMetadata.LatestVersion, latestProcessModuleConfigCache); err != nil {
```

Comment:
assignment and condition should be split up for better readability. I almost missed that there was an if when reviewing